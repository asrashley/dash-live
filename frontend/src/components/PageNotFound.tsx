import { routeMap, uiRouteMap } from "@dashlive/routemap";
import { Card } from "./Card";

export function PageNotFound() {
  const imageSrc = routeMap.images.url({ filename: "sad-puppy-dog.svg" });
  return (
    <Card
      id="page-not-found"
      className="m-3"
      header="Sorry I don't know about this page"
    >
      <div className="d-flex flex-row mb-3">
        <div>
          <img
            src={imageSrc}
            alt="Vector image of a dog looking sad"
            className="not-found-img"
          />
          <a
            className="fs-6 link-opacity-50"
            href="https://www.wannapik.com/vectors/88"
          >
            Image designed by Wannapik
          </a>
        </div>
        <div>
          <p className="fs-3">
            This page might have moved, or it might be a bug in this site.
          </p>
          <p className="fs-4">
            Probably a good idea to{" "}
            <a href={uiRouteMap.home.url()} className="link link-underline-light">
              return to the home page
            </a>
            .
          </p>
        </div>
      </div>
    </Card>
  );
}
