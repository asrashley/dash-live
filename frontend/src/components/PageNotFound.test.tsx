import { describe, expect, test } from "vitest"
import { renderWithProviders } from "../test/renderWithProviders"
import { PageNotFound } from "./PageNotFound";

describe('PageNotFound component', () => {
    test('matches snapshot', () => {
        const { asFragment } = renderWithProviders(<PageNotFound />);
        expect(asFragment()).toMatchSnapshot();
    });
});