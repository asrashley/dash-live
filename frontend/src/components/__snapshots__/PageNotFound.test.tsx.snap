// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNotFound component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card m-3"
    id="page-not-found"
  >
    <div
      class="card-header"
    >
      Sorry I don't know about this page
    </div>
    <div
      class="card-body"
    >
      <div
        class="d-flex flex-row mb-3"
      >
        <div>
          <img
            alt="Vector image of a dog looking sad"
            class="not-found-img"
            src="/static/img/sad-puppy-dog.svg"
          />
          <a
            class="fs-6 link-opacity-50"
            href="https://www.wannapik.com/vectors/88"
          >
            Image designed by Wannapik
          </a>
        </div>
        <div>
          <p
            class="fs-3"
          >
            This page might have moved, or it might be a bug in this site.
          </p>
          <p
            class="fs-4"
          >
            Probably a good idea to 
            <a
              class="link link-underline-light"
              href="/"
            >
              return to the home page
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
